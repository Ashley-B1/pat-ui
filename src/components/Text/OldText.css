/* * {
  margin:0;
  padding:0;
} */
/* .App {
    font-family: sans-serif;
    background-color: #c7c7c7;
    position: absolute;
    width: 100%;
    min-height: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
  } 
  .App > p,
  h1 {
    margin-left: 50%;
  }
  .App > h3 {
    padding: 5% 5% 0;
  }
  .container {
    position: relative;
    padding: 0 10%;
  } */
  .textbox-container {
    /* position: relative; */
    /* width: 100%; */
    display: flex;
    /* align-items: center;
    justify-content: center; */
  }
  .textbox {
    /* outline: none; */
    outline: 1px solid rgb(196, 196, 196);

    border: none;
    background-color: white;
    /*  */
    height: 30px;
    padding: 1rem 1rem;
  }
  .textbox:hover {
    outline: 2px solid gray;
  }
  .textbox:focus{
    outline: 2px solid black;
  }
  .isError {
    background-color: #ffe0e0;
  }
  .textbox.isError:focus {
    outline: 2px solid #990022;
    background-color: white;
  }
  
  .svg-error {
    color: red;
    /* position: absolute; */
    position: relative;
    display: inline-block;
    width: 20px;
    height: 20px;
    top:5px;
    right: 1.5rem;
  }