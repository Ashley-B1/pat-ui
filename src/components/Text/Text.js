import React,  { useState } from 'react'

function Text({ vaild = true, value, placeholder, ...rest }) {
  const [inputValue, setInputValue] = useState(value);

  return (
    <input
      vaild={vaild}
      className={`${vaild ? "input-box" : "error-box"}`}
      type="text"
      value={inputValue}
      placeholder={placeholder}
      onChange={(e) => {
        setInputValue(e.target.value);
      }}
      {...rest}
    ></input>
  );
}
  

export default Text;