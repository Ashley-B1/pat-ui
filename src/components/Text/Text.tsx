import React, { FC, ReactNode, useState, InputHTMLAttributes } from 'react';
import { classNames } from '../../utils/classNames';


interface ITextProps {
    /** set message content */
    value?: string;
    /** set message list */
    vaild?: boolean;
    /** set message icon */
    placeholder?: string;
    /** set message click action */
    onChange?: () => void;
}

export type JTextProps = ITextProps & InputHTMLAttributes<HTMLInputElement>;

export const Text: FC<JTextProps> = (props) => {
    const { 
        vaild, 
        value, 
        placeholder,
        onChange,
        ...rest } = props;

    const [inputValue, setInputValue] = useState(value);

    let text = (
            <input
              className={`${vaild ? "input-box" : "error-box"}`}
              type="text"
              value={inputValue}
              placeholder={placeholder}
              onChange={(e) => {
                setInputValue(e.target.value);
              }}
              {...rest}
            ></input>
    );

    return text;
};

Text.defaultProps = {
    type: 'text',
    vaild: true,
};

export default Text;