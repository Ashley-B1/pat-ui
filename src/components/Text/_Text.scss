.input-container {
  position: relative;
  display: flex;
  flex-direction: row;
  @include text-box(3rem, 10rem, $font-size-base, 1rem);
  .input-text {
    border: none;
    @include text-style(
      0.5rem,
      $gray-900,
      $white,
      $gray-200,
      $gray-500,
      $gray-900,
      $white,
      0.3rem
    );
  }
  .errIcon {
    fill: red;
    position: absolute;
    top: 50%;
    right: 0;
    transform: translate(100%, -50%);
  }
}

.with-error {
  .input-text {
    @include text-style(
      0.5rem,
      $gray-900,
      $msg-pink-background,
      $msg-pink-background,
      $gray-400,
      $pink,
      $white,
      0.3rem
    );
  }
}
