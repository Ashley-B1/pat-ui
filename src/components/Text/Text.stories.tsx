import React from 'react';
import { action } from '@storybook/addon-actions';
import Text from './Text';

export default {
  title: 'Text',
  component: Text,
};
export const DefaultText = () => (
  <div>
    <Text className=""></Text>
    <Text placeholder="Placeholder"></Text>
    <Text value="Input"></Text>
  </div>
);
export const ErrorText = () => (
  <div>
    <Text invalid={true}></Text>
    <Text invalid={true} placeholder="Placeholder"></Text>
    <Text invalid={true} value="Input"></Text>
  </div>
);
