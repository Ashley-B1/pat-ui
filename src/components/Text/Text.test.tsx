
import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import Text, {  } from './Text';

describe('Text', () => {
//   it('should match snapshot', () => {
//     const { asFragment } = render(<Text> Snapshot Text </Text>);
//     expect(asFragment()).toMatchSnapshot();
//   });

  it('should render default text', () => {
    const txtProps = {
      onClick: jest.fn(),
    };
    const wrapper = render(<Text {...txtProps}>Default Text</Text>);
    const element = wrapper.queryByText('Default Text') as HTMLElement;
    expect(element).toBeInTheDocument();
    expect(element.tagName).toBe('TEXT');
    expect(txtProps.onClick).toHaveBeenCalledTimes(0);
    fireEvent.click(element);
    expect(txtProps.onClick).toHaveBeenCalledTimes(1);
  });

  
});