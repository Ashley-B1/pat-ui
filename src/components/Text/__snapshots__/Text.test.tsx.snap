// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<DocumentFragment>
   <div class="inputContainer">
        <input></input>
        <img />
   </div>
</DocumentFragment>
`;
